examples/patterns/batch-processing.ts

```
@@ -0,0 +1,35 @@
+const queue = new LinkedQueue<Task>();
+
+for (let i = 1; i <= 12; i++) {
+  queue.offer({ id: i, name: `Task ${i}` });
+}
+
+const batchSize = 5;
+const batch = new ArrayList<Task>();
+
+let task: Task | undefined;
+
+while ((task = queue.dequeue()) !== undefined) {
```

Biome (2.4.15)

[error] The assignment should not be in an expression.

(lint/suspicious/noAssignInExpressions)